### PR TITLE
[pylint] delete return operation

### PR DESCRIPTION
**Pull request by l0lawrence**
Current only instances of Client methods for:

def delete_* -> Keyvault-Certs, Search, Storage File share/datalake etc.
def begin_delete -> keyvault : "LROPoller[DeletedKey]"

A lot of exceptions to this rule ^^ as above

### CODE REVIEW

File: tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py [hunk 1]
```diff
@@ -2028,6 +2028,47 @@ def _check_import(self, name, node):
...
 
+class DeleteOperationReturnStatement(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    """Rule to check that a delete* or begin_delete* returns None or LROPoller[None]."""
+    name = "delete-operation-wrong-return-type"
+    priority = -1
+    msgs = {
+        "C4752": (
+            "delete* or begin_delete* does not return None or LROPoller[None].",
```
Comment (kristapratico) [inline tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py:2039]: nit: wording

```suggestion
            "delete* or begin_delete* should return None or LROPoller[None], respectively.",
```

File: tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py [hunk 2]
```diff
@@ -2028,6 +2028,47 @@ def _check_import(self, name, node):
...
+    def visit_functiondef(self,node):
+        """Visits all delete functions and checks that their return types
+        are LROPoller or None. """
+        try:
+            if node.returns.as_string() == "None":
+                # If there are residual comment typehints or no return value,
+                # we dont want to throw an error
+                return
+            if node.name.startswith("delete") and node.parent.name.endswith("Client"):
+                if node.returns.as_string() != "None":
```
Comment (kristapratico) [inline tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py:2054]: does `node.returns` access the type hints?

File: tools/pylint-extensions/pylint-guidelines-checker/tests/test_pylint_custom_plugins.py [hunk 3]
```diff
@@ -3136,4 +3136,31 @@ def test_raise_traceback(self):
...
-            self.checker.visit_importfrom(node)
\ No newline at end of file
+            self.checker.visit_importfrom(node)
+
+class TestDeleteOperationReturnType(pylint.testutils.CheckerTestCase):
+
+    """Test that we are checking the return type of delete functions is correct"""
+    CHECKER_CLASS = checker.DeleteOperationReturnStatement
+
+    def test_delete_operation(self):
```
Comment (kristapratico) [inline tools/pylint-extensions/pylint-guidelines-checker/tests/test_pylint_custom_plugins.py:3146]: looks like we need more tests when you get a chance :)

File: tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py (hunk 2 above)
Comment (l0lawrence) [reply tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py:2054]: it accesses the return annotation, there is also type_comment_returns that accesses the return type from a type comment.

**Review by kristapratico — APPROVED** [review]

**Pull request merged**